doc/rst/language/spec/procedures.rst

```diff
@@ -586,95 +586,51 @@ modify the formal argument within its dynamic scope. Whether ``const``
 is interpreted as ``const in`` or ``const ref`` intent depends on the
 argument type.  Generally, small values, such as scalar types, will be
 passed by ``const in``; while larger values, such as domains and
-arrays, will be passed by ``const ref`` intent. The
-:ref:`Abstract_Intents_Table` below lists the meaning of the ``const``
-intent for each type.
+arrays, will be passed by ``const ref`` intent.
+
+For the following types, the ``const`` intent means ``const in``:
+
+  * scalar types (``bool``, ``int``, ``uint``, ``real``, ``imag``, ``complex``)
+  * ranges
+  * non-managed classes (``borrowed``, ``unmanaged``)
+
+For the following types, the ``const`` intent means ``const ref``:
+
+  * string-like types (``string``, ``bytes``)
+  * domains / domain maps
+  * arrays
+  * records
+  * auto-managed classes (``owned``, ``shared``) - see :ref:`Default_Intent_for_owned_and_shared`
+  * unions ``const ref``
+  * synchronization types (``atomic``, ``sync``)
+
+The ``const`` intent for tuples applies the ``const`` intent to each tuple
+component as if it was passed as a separate argument.
+See :ref:`Tuple_Argument_Intents`.
```

Comment:
Is that completely accurate? I would think that a tuple containing a sync variable would have that component be mutable. But, I'm not so sure which of these we should do. Would you be willing to open an issue about it?

Reply:
Well there seems to be many problems here, I will definitely be opening an issue.

This program compiles and runs today

```chapel

proc defaultIntent(tup: (int, sync int)) {
  tup(0) = 1; // this definitly shouldn't work
  tup(1).writeEF(5); // this works
}
proc constIntent(const tup: (int, sync int)) {
  // tup(0) = 1; // this definitly shouldn't work
  tup(1).writeEF(10); // This probably shouldn't work
}

var x: (int, sync int);
x(1).writeEF(2);
writeln(x(0), " ", x(1).readFE());
defaultIntent(x);
writeln(x(0), " ", x(1).readFE());
constIntent(x);
writeln(x(0), " ", x(1).readFE());

proc constSync(const x: sync int) {
  x.writeEF(10); // this definitly should not work
}
var y: sync int;
constSync(y);
writeln(y.readFE());
```

Reply:
My way out here is to change this to
```
Tuples have special handling of the ``const`` intent, see :ref:`Tuple_Argument_Intents`.
```
Definitely looking for better wording here

Reply:
IMO what you have is good enough for now.

Reply:
I will leave it alone then. I opened #23416 and #23413 for the tuples